From 074bbd80d28c3ca6cc1a06771e35685d80b8e9a2 Mon Sep 17 00:00:00 2001
Message-Id: <074bbd80d28c3ca6cc1a06771e35685d80b8e9a2.1430272235.git.chang-joon.lee@intel.com>
In-Reply-To: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
References: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Fri, 10 Apr 2015 13:41:42 +0530
Subject: [PATCH 02/10] FOR_UPSTREAM [VPG]: drm/i915: fix flis dsi rcomp
 programming

All the bits values except the lower nibble is preserved
while programming this register.

Issue: GMINL-8141
Change-Id: I3180d437b45860c56885982cf479fb6a1795e089
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |    7 ++++---
 drivers/gpu/drm/i915/intel_dsi.h |    2 ++
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 2a46d8e..95389a5 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -194,9 +194,10 @@ static void intel_dsi_device_ready(struct intel_encoder *encoder)
 	DRM_DEBUG_KMS("\n");
 
 	mutex_lock(&dev_priv->dpio_lock);
-	/* program rcomp for compliance, reduce from 50 ohms to 45 ohms
-	 * needed everytime after power gate */
-	vlv_flisdsi_write(dev_priv, 0x04, 0x0004);
+
+	val = vlv_flisdsi_read(dev_priv, FLIS_RCOMP_IOSFSB_REG4);
+	val &= ~(0x0f);
+	vlv_flisdsi_write(dev_priv, FLIS_RCOMP_IOSFSB_REG4, val);
 	mutex_unlock(&dev_priv->dpio_lock);
 
 	/* bandgap reset is needed after everytime we do power gate */
diff --git a/drivers/gpu/drm/i915/intel_dsi.h b/drivers/gpu/drm/i915/intel_dsi.h
index e00ce5f..a943dee 100644
--- a/drivers/gpu/drm/i915/intel_dsi.h
+++ b/drivers/gpu/drm/i915/intel_dsi.h
@@ -397,6 +397,8 @@
 
 #define LPIO_PWM_ENABLE_MASK		0x80000000
 
+#define FLIS_RCOMP_IOSFSB_REG4		0x04
+
 struct intel_dsi_device {
 	unsigned int panel_id;
 	const char *name;
-- 
1.7.9.5

